src/tools/illink/test/Mono.Linker.Tests.Cases/CommandLine/NoSEHFlagIsSet.cs

```diff
@@ -0,0 +1,24 @@
+using Mono.Linker.Tests.Cases.Expectations.Assertions;
+using Mono.Linker.Tests.Cases.Expectations.Metadata;
+
+namespace Mono.Linker.Tests.Cases.CommandLine
+{
+    /// <summary>
+    /// This test case verifies that the NoSEH flag is set in the PE header
```

Comment:
This test would have to produce R2R published app to actually test the scenario.

I am not sure whether it is worth the troubles.